fe/fe-core/src/main/java/org/apache/doris/planner/DistributedPlanner.java

```diff
@@ -268,7 +268,6 @@ private PlanFragment createMergeFragment(PlanFragment inputFragment)
         mergePlan.init(ctx.getRootAnalyzer());
         Preconditions.checkState(mergePlan.hasValidStats());
         PlanFragment fragment = new PlanFragment(ctx.getNextFragmentId(), mergePlan, DataPartition.UNPARTITIONED);
-        fragment.setResultSinkType(ctx.getRootAnalyzer().getContext().getResultSinkType());
```

Comment:
why modify this code?

Reply:
Because the `fragment` here in some cases is not `root fragment`, so after `createFragments`, set ResultSinkType to `root fragment`.